provider/k8s.go

```diff
@@ -659,6 +661,149 @@ func (store genericFileStore) NumRows(key string) (int64, error) {
 	}
 }
 
+type csvIterator struct {
+	reader        *csv.Reader
+	currentValues GenericRecord
+	err           error
+	columnNames   []string
+	idx           int64
+	limit         int64
+}
+
+func (c *csvIterator) Next() bool {
+	if c.idx >= c.limit {
+		return false
+	}
+	row, err := c.reader.Read()
+	if err != nil {
+		if err == io.EOF {
+			return false
+		} else {
+			c.err = err
+			return false
+		}
+	}
+	c.currentValues = c.ParseRow(row)
+	c.idx += 1
+	return true
+}
+
+func (c *csvIterator) Values() GenericRecord {
+	return c.currentValues
+}
+
+func (c *csvIterator) Columns() []string {
+	return c.columnNames
+}
+
+func (c *csvIterator) Err() error {
+	return c.err
+}
+
+func (c *csvIterator) Close() error {
+	return nil
+}
+
+func (c *csvIterator) ParseRow(row []string) GenericRecord {
+	records := make(GenericRecord, len(row))
+	for i, value := range row {
+		if integer, err := strconv.Atoi(value); err == nil {
```

Comment:
Does this only parse ints a floats?

Reply:
Yes, I can extend this but I figured these basic types could cover the most ground.

Reply:
Would recommend adding string at least since thats common

Reply:
may need to change this format. seems like it with either always be cast as an int or error

Reply:
Sorry, maybe this wasn't clear: if it can't part an int or a float, it'll keep the value as a string.

Reply:
Ah i see, i was reading it backwards